source/euclid.js

```diff
@@ -0,0 +1,21 @@
+// Макаров Никита WEB-22
+
+'use strict';
+
+const greatestCommonDivisor = (a, b) => (b === 0 ? a : greatestCommonDivisor(b, a % b));
+
+const validateNaturalNumbers = numbers => {
+    return numbers.every(number => Number.isInteger(number) && number > 0) && !(numbers.length === 0);
```

Comment:
@DeadBread-001, еще стоит добавить тест на это и вместо `!(numbers.length === 0)` есть более простая запись:

```suggestion
    return numbers.every(number => Number.isInteger(number) && number > 0) && numbers.length !== 0;
```

Тем не менее, эта не проверка на `undefined` :)

Reply:
Запись упрощу, но есть вопрос. Всё-таки если подать в функцию undefined либо вообще ничего не вводить, то это попадает под условия в проверке на валидность. Или лучше стоит добавить `numbers !== undefined`?